### ISSUE TITLE
Test set contains nodes that are not present in the training graph

### ISSUE THREAD

**Issue by nomisto** [OpenBioLink/OpenBioLink#40]
The training set doesn't get filtered from the test set after the train-validation set is split up, so there can be nodes in the test set that aren't in the training set, since they are in the validation set.
One fix would be to put all the filtering at the end of the split function.
